### PR TITLE
feat(dlp): added a sample for de-identify exception list

### PR DESCRIPTION
**Pull request by AarshD-crest**
- Added a sample for [de-identify exception list](https://cloud.google.com/dlp/docs/samples/dlp-deidentify-exception-list) with unit test-case.

### CODE REVIEW

File: dlp/snippets/deid/deid_with_exception_list.go [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+	"context"
+	"fmt"
+	"io"
+
+	dlp "cloud.google.com/go/dlp/apiv2"
+	"cloud.google.com/go/dlp/apiv2/dlppb"
+)
+
+// deidentifyExceptionList creates an exception list for a regular custom dictionary detector.
+func deidentifyExceptionList(w io.Writer, projectID, input, infoType string, dictWordList []string) error {
```
Comment (telpirion) [inline dlp/snippets/deid/deid_with_exception_list.go:27]: nit: you can reduce the input to just what's needed for testing (i.e. `projectID`). 

See here:
https://googlecloudplatform.github.io/samples-style-guide/#minimal-arguments

File: dlp/snippets/deid/deid_with_exception_list.go [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+	client, err := dlp.NewClient(ctx)
+	if err != nil {
+		return fmt.Errorf("dlp.NewClient: %v", err)
+	}
+
+	// Closing the client safely cleans up background resources.
+	defer client.Close()
+
+	// Specify what content you want the service to DeIdentify.
+	var item = &dlppb.ContentItem{
```
Comment (telpirion) [inline dlp/snippets/deid/deid_with_exception_list.go:47]: issue: remove `var` and use `:=` operator.

File: dlp/snippets/deid/deid_with_exception_list.go [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+		DataItem: &dlppb.ContentItem_Value{
+			Value: input,
+		},
+	}
+
+	// Specify the word list custom info type and build-in info type the inspection will look for.
+	var infoTypes = []*dlppb.InfoType{
```
Comment (telpirion) [inline dlp/snippets/deid/deid_with_exception_list.go:54]: issue: remove `var` and use `:=` operator.

File: dlp/snippets/deid/deid_with_exception_list.go [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+		Rules: []*dlppb.InspectionRule{
+			{
+				Type: &dlppb.InspectionRule_ExclusionRule{
+					ExclusionRule: &dlppb.ExclusionRule{
+						MatchingType: dlppb.MatchingType_MATCHING_TYPE_FULL_MATCH,
+						Type: &dlppb.ExclusionRule_Dictionary{
+							Dictionary: &dlppb.CustomInfoType_Dictionary{
+								Source: &dlppb.CustomInfoType_Dictionary_WordList_{
+									WordList: &dlppb.CustomInfoType_Dictionary_WordList{
+										Words: dictWordList,
```
Comment (telpirion) [inline dlp/snippets/deid/deid_with_exception_list.go:69]: nit: this nesting gets a bit deep. Consider breaking out the deepest level(s) into separate lines.

File: dlp/snippets/deid/deid_with_exception_list.go [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	// Construct the configuration for the de-id request and list all desired transformations.
+	var deIdentifyConfig = &dlppb.DeidentifyConfig{
```
Comment (telpirion) [inline dlp/snippets/deid/deid_with_exception_list.go:81]: issue: remove `var` and use `:=` operator.

File: dlp/snippets/deid/deid_with_exception_list.go [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+			RuleSet:         []*dlppb.InspectionRuleSet{inspectRuleSet},
+		},
+		// The item to analyze.
+		Item: item,
+	}
+
+	// Send the request.
+	resp, err := client.DeidentifyContent(ctx, req)
+	if err != nil {
+		return fmt.Errorf("DeidentifyContent: %v", err)
```
Comment (telpirion) [inline dlp/snippets/deid/deid_with_exception_list.go:111]: issue: return the error directly to the caller.

Comment (AarshD-crest) [reply dlp/snippets/deid/deid_with_exception_list.go]: fixed it.

File: dlp/snippets/deid/deid_with_exception_list.go (hunk 5 above)
Comment (AarshD-crest) [reply dlp/snippets/deid/deid_with_exception_list.go]: fixed it.

File: dlp/snippets/deid/deid_with_exception_list.go (hunk 4 above)
Comment (AarshD-crest) [reply dlp/snippets/deid/deid_with_exception_list.go]: got that, changed it.

File: dlp/snippets/deid/deid_with_exception_list.go (hunk 3 above)
Comment (AarshD-crest) [reply dlp/snippets/deid/deid_with_exception_list.go]: fixed it.

File: dlp/snippets/deid/deid_with_exception_list.go (hunk 2 above)
Comment (AarshD-crest) [reply dlp/snippets/deid/deid_with_exception_list.go]: fixed it.

File: dlp/snippets/deid/deid_with_exception_list.go (hunk 1 above)
Comment (AarshD-crest) [reply dlp/snippets/deid/deid_with_exception_list.go]: got that, and changed it.

File: dlp/snippets/deid/deid_test.go [hunk 7]
```diff
@@ -100,3 +100,25 @@ func TestDeidentifyDateShift(t *testing.T) {
...
+
+func TestDeidentifyExceptionList(t *testing.T) {
+	tc := testutil.SystemTest(t)
+
+	input := "jack@example.org accessed customer record of user5@example.com"
+	infoType := "EMAIL_ADDRESS"
+	dictWordList := []string{"jack@example.org", "jill@example.org"}
+	want := "output : jack@example.org accessed customer record of [EMAIL_ADDRESS]"
+
+	t.Run(input, func(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (AarshD-crest) [reply dlp/snippets/deid/deid_test.go]: fixed it.

**Review by telpirion — APPROVED** [review]

**Pull request merged**